src/main/java/nl/ycn/coaching/controller/HrController.java

```diff
@@ -0,0 +1,85 @@
+package nl.ycn.coaching.controller;
+
+
+import net.bytebuddy.utility.RandomString;
+import nl.ycn.coaching.database.AppUserService;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Controller
+public class HrController {
+
+	private String randomPassword="";
+
+	@Autowired
+	private AppUserService appUserService;
+
+	//Mappings for HR-Employee
+	@GetMapping("bootcamps")
+	public String getBootcamps(){return "hremployee/bootcamps";}
+
+	@GetMapping("users")
+	public String getUsers(){return "hremployee/users";}
+
+	@GetMapping("teams")
+	public String getTeams(){return "hremployee/teams";}
+
+	@GetMapping("skills")
+	public String getSkills(){return "hremployee/skills";}
+
+	@GetMapping("createsoftskillform")
+	public String getSoftskillForm(){return "hremployee/createsoftskillform";}
+
+	@GetMapping("/register")
+	public String register(Model model){
+		model.addAttribute(randomPassword, randomPassword);
+		return "/hremployee/register";
+	}
+
+	@GetMapping("hrcalendar")
+	public String getCalendar(){
+		return "hremployee/calendar";}
+
+	//register a new AppUser (from hremployee/register)
+	@PostMapping("/register")
+	public String register(String username, String firstname, String lastname, String email, String password, String roles){
+
+		BCryptPasswordEncoder encoder = new BCryptPasswordEncoder();
+		appUserService.registerUser(username,  firstname, lastname, email, encoder.encode(password), roles);
+		randomPassword = "";
+		return "/hremployee/users";
+	}
+
+	//generate a random password for a new AppUser (from hremployee/register)
+	@GetMapping("getpassword")
```

Comment:
Als je een methode wilt aanroepen vanuit thymeleaf, en niet een nieuwe template pagina wilt laden, kijk eens naar @RequestMapping om json terug te geven, in plaats van @GetMapping